prow/spyglass/lenses/metadata/lens.go

```diff
@@ -137,6 +146,79 @@ func (lens Lens) Body(artifacts []lenses.Artifact, resourceDir string, data stri
+
+func hintFromPodInfo(buf []byte) string {
+	var report k8sreporter.PodReport
+	if err := json.Unmarshal(buf, &report); err != nil {
+		// This error isn't worth highlighting here, and will come up again
+		// elsewhere.
+		return ""
+	}
+
+	// We're more likely to pick a relevant event if we use the last ones first.
+	sort.Slice(report.Events, func(i, j int) bool {
+		a := &report.Events[i]
+		b := &report.Events[j]
+		return b.LastTimestamp.Before(&a.LastTimestamp)
+	})
+
+	// If the pod completed successfully there's probably not much to say.
+	if report.Pod.Status.Phase == v1.PodSucceeded {
+		return ""
+	}
+	// Check if we have any images that didn't pull
+	for _, s := range append(report.Pod.Status.InitContainerStatuses, report.Pod.Status.ContainerStatuses...) {
+		if s.State.Waiting != nil && s.State.Waiting.Reason == "ImagePullBackOff" {
+			return fmt.Sprintf("The %s container could not start because it could not pull %q. Check your images.", s.Name, s.Image)
+		}
+	}
+	// Check if we're trying to mount a volume
+	if report.Pod.Status.Phase == v1.PodPending {
+		failedMount := false
+		for _, e := range report.Events {
+			if e.Reason == "FailedMount" {
+				failedMount = true
+				if strings.HasPrefix(e.Message, "MountVolume.SetUp") {
+					// Annoyingly, parsing this message is the only way to get this information.
+					// If we can't parse it, we'll fall through to a generic bad volume message below.
+					results := failedMountRegex.FindStringSubmatch(e.Message)
+					if results == nil {
+						continue
+					}
+					return fmt.Sprintf("The pod could not start because it could not mount the volume %q: %s", results[1], results[2])
+				}
+			}
+		}
+		if failedMount {
+			return fmt.Sprintf("The job could not start because one or more of the volumes could not be mounted.")
+		}
+	}
+	// Check if we cannot be scheduled
+	// This is unlikely - we only outright fail if a pod is actually scheduled to a pod it does not support
+	if report.Pod.Status.Phase == v1.PodFailed && report.Pod.Status.Reason == "MatchNodeSelector" {
+		return fmt.Sprintf("The job could not start because it was scheduled to a node that does not satisfy its NodeSelector")
+	}
+	// Usually we would fail to schedule it at all, so it will be pending forever.
+	if report.Pod.Status.Phase == v1.PodPending {
+		for _, e := range report.Events {
+			if e.Reason == "FailedScheduling" {
+				return fmt.Sprintf("There are no nodes that your pod can schedule to - check your requests, tolerations, and node selectors (%s)", e.Message)
+			}
+		}
+	}
+
+	// There are a bunch of fun ways for the node to fail that we've seen before
+	for _, e := range report.Events {
+		if e.Reason == "FailedCreatePodSandbox" || e.Reason == "FailedSync" || e.Reason == "FailedKillPod" {
+			return "The job may have executed on an unhealthy node. Contact your prow maintainers or check the detailed pod information."
+		}
+	}
+
+	// We've got nothing.
+	return ""
```

Comment:
We could perhaps change this to be something like

```
No known issues were found by inspecting the job resources. <some blurb about how to constructively add new cases or make an issue to track new cases>
```

Reply:
In almost all cases when a job fails, it's because the test failed. I don't think it's useful for us to suggest a job issue or infrastructure failure in the UI in the case where we don't see anything obviously wrong - it'll mostly just be because their code was bad.

Reply:
Sorry, I thought we'd hit this case when the obvious case (test failed) was not what happened. I understand your point